tests/study_tests/test_optimize.py

```diff
@@ -1,49 +1,188 @@
+from typing import Callable
+from typing import Optional
+from unittest import mock
+
+from _pytest.logging import LogCaptureFixture
 import pytest
 
 from optuna import create_study
+from optuna import logging
 from optuna import Trial
+from optuna import TrialPruned
 from optuna.study import _optimize
+from optuna.study._tell import _tell_with_warning
 from optuna.testing.storage import STORAGE_MODES
 from optuna.testing.storage import StorageSupplier
 from optuna.trial import TrialState
 
 
 @pytest.mark.parametrize("storage_mode", STORAGE_MODES)
-def test_run_trial(storage_mode: str) -> None:
+def test_run_trial(storage_mode: str, caplog: LogCaptureFixture) -> None:
+    # We need to reconstruct our default handler to properly capture stderr.
+    logging._reset_library_root_logger()
+    logging.enable_default_handler()
+    logging.set_verbosity(logging.INFO)
+    logging.enable_propagation()
```

Comment:
Can we do this, and the disabling of the propagation, in a `pytest.fixture` https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization ?

Reply:
Ah, thank you for letting me know! I extract this setup/teardown to the fixture!
5c199a7